sql_metadata/parser.py

```diff
@@ -1,3 +1,4 @@
+# pylint: disable=C0302
```

Comment:
> Used when a module has more lines than the limit specified in the max-module-lines option.

Shouldn't we refactor this quite huge module instead of ignoring the linter? Let's do it in a separate PR.

Reply:
Yes that's what I though too - refactor in separate pr.

It again became quite complex as it used to be - it's a "Structured" Query language but it's damned complicated to parse 🤣  

I was thinking about maybe adding `token_type` (or even subclasses and polymorphism) to our token and assigning a type for each token. That way it should speed up the parsing as one token cannot have two functions in particular query (if it's an alias it cannot be a table etc.) so we can skip all tokens that already have a type in all methods. That probably will simplify also some of the checks as we are repeating checks on the same tokens all the time but in different context/functions.

Reply:
Makes sense, let's tackle this one in a separate PR or even discuss them earlier on GitHub.